x-pack/plugins/cases/common/types/domain/configure/v1.ts

```diff
@@ -7,20 +7,31 @@
 
 import * as rt from 'io-ts';
 import { CaseConnectorRt, ConnectorMappingsRt } from '../connector/v1';
+import { TextCustomFieldRt, ToggleCustomFieldRt } from '../custom_field/v1';
 import { UserRt } from '../user/v1';
 
 const ClosureTypeRt = rt.union([rt.literal('close-by-user'), rt.literal('close-by-pushing')]);
 
-export const ConfigurationBasicWithoutOwnerRt = rt.strict({
-  /**
-   * The external connector
-   */
-  connector: CaseConnectorRt,
-  /**
-   * Whether to close the case after it has been synced with the external system
-   */
-  closure_type: ClosureTypeRt,
-});
+export const ConfigurationBasicWithoutOwnerRt = rt.intersection([
+  rt.strict({
+    /**
+     * The external connector
+     */
+    connector: CaseConnectorRt,
+    /**
+     * Whether to close the case after it has been synced with the external system
+     */
+    closure_type: ClosureTypeRt,
+  }),
+  rt.exact(
```

Comment:
I think that the custom fields in the domain type should be required. In `x-pack/plugins/cases/common/types/api/configure/v1.ts` the requests should not have them required. We would need to set the custom fields to an empty array before saving them to ES.